### PR TITLE
[Enhancement] Add Citation Meta Data

### PR DESCRIPTION
**Pull request by kevinmatthes**
This Pull Requests introduces initial citation meta data for this project using the [Citation File Format](https://github.com/citation-file-format/citation-file-format) standard.  I also added the automatic validator such that one can rely on the correctness of the configuration file.

The introduced citation meta data are rendered by GitHub to a "Cite this repository" option on this project's welcome page.  People interested in this project can then cite it in their papers and theses.

The meta data can be freely enriched with further fields as required.  Furthermore, there is the option to add a list of references to it.  This way, you can cite the used resources in order to create this project.

Please tell me whether the suggested initial meta data set is correct and if I should add further entries.  For instance, there are the options to add the crate's description on crates.io as `repository-artifact` or to alter the URL to the documentation website.  I decided to leave these options open in order to discuss the other entries, at first.

### CODE REVIEW

File: .github/workflows/cffconvert.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+permissions:
+  contents: read
+
+jobs:
+  validation:
+    name: validate
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
```
Comment (sorairolake) [inline .github/workflows/cffconvert.yml:23]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/cffconvert.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          persist-credentials: false
+
+      - name: cffconvert
+        uses: citation-file-format/cffconvert-github-action@4cf11baa70a673bfdf9dad0acc7ee33b3f4b6084
```
Comment (sorairolake) [inline .github/workflows/cffconvert.yml:28]:
```suggestion
        uses: citation-file-format/cffconvert-github-action@2.0.0
```

File: .github/workflows/cffconvert.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    paths:
+      - CITATION.cff
+
+permissions:
+  contents: read
+
+jobs:
+  validation:
+    name: validate
+    runs-on: ubuntu-latest
```
Comment (sorairolake) [inline .github/workflows/cffconvert.yml:20]:
```suggestion
    runs-on: ubuntu-20.04
```

**Review by sorairolake — COMMENTED** [review]
Thank you for the PR. I don't know about this, but I think this is fine.

**Comment by sorairolake** [comment]
Also, could you change the extension from `.yml` to `.yaml` if possible?

**Comment by kevinmatthes** [comment]
Okay, I applied all requested changes. :wrench: :heavy_check_mark:

**Review by sorairolake — APPROVED** [review]

**Comment by sorairolake** [comment]
Merged this. Thanks!

**Pull request merged**